srcs/app/http/HttpRequestHandler.cpp

```diff
@@ -36,75 +68,206 @@ void		HttpRequestHandler::HandleRequest_() {
+	struct dirent *entry;
+	while ((entry = readdir(dir)) != NULL) {
+		std::string name = entry->d_name;
+		if (name.rfind(".", 0) == 0) {
+			continue;
+		}
+		const std::string full_path_name = full_path + "/" + name;
+		if (IsDirectory_(full_path_name)) {
+				name.append("/");
+		}
+		*body << "<a href=\"" << name << "\">" << name << "</a>\n";
+	}
+	closedir(dir);
+	return true;
+}
+
+void	HttpRequestHandler::ListDirectory_(const std::string &request_path) {
+	std::stringstream body;
+	body << "<html>\n" <<
+		"<head><title>Index of " << request_path << "</title></head>\n" <<
+		"<body>\n" <<
+		"<h1>Index of " << request_path <<
+		"</h1><hr><pre><a href=\"../\">../</a>\n";
+	const std::string full_path = request_location_->common.root + request_path;
+	if (!TryAddDirectoryContent_(&body, full_path)) {
+		return;
+	}
+	body << "</pre><hr></body>\n" <<
+		"</html>\n";
 	HttpResponse	response(200);
-	std::string		body = "Responding to a GET request\n";
+	response.AddHeader("Content-Type", "text/html");
+	response.SetBody(body.str());
+	AddCommonHeaders_(&response);
+	raw_response_ = response.CreateResponseString();
+}
 
-	response.AddHeader("Content-Type", "text/plain");
+bool	HttpRequestHandler::HasAcceptedFormat_(const HttpRequest &request) {
+	if (request_location_->HasLocation() &&
+									!request_location_->limit_except->empty()) {
+		if (std::find(request_location_->limit_except->begin(),
+				request_location_->limit_except->end(),
+				request.GetMethod()) ==
+									request_location_->limit_except->end()) {
+			RequestError_(405);
+			return false;
+		}
+	}
+	if (request.GetBody().size() >
+							request_location_->common.client_max_body_size) {
+		RequestError_(413);
+		return false;
+	}
+	if (request.HasHeader("Content-Encoding")) {
+		RequestError_(415);
+		return false;
+	}
+	return true;
+}
+
+void	HttpRequestHandler::ServeFile_(const std::string &file_path) {
+	if (!IsRegularFile_(file_path)) {
+		RequestError_(403);
+		return;
+	}
+	std::ifstream ifs(file_path.c_str(), std::ios::in | std::ios::binary);
+	if (!ifs) {
+		RequestError_(404);
+		return;
+	}
+	const std::string body = std::string(std::istreambuf_iterator<char>(ifs),
+											std::istreambuf_iterator<char>());
+	HttpResponse response(200);
+	AddCommonHeaders_(&response);
+	response.AddHeader("Content-Type", GetMimeType_(file_path));
 	response.SetBody(body);
+	raw_response_ = response.CreateResponseString();
+}
+
+void	HttpRequestHandler::MovedPermanently_(const HttpRequest &request) {
+	HttpResponse response(301);
 	AddCommonHeaders_(&response);
+	response.AddHeader("Content-Type", "text/html");
+	std::stringstream url;
+	url << "http://" << request.GetHost() << ":" << request.GetPort() <<
+		request.GetPath() << "/";
+	response.AddHeader("Location", url.str());
+	const std::string body = DefaultResponseBody_(301);
+	response.SetBody(body);
 	raw_response_ = response.CreateResponseString();
 }
 
+void	HttpRequestHandler::DoGet_(const HttpRequest &request) {
```

Comment:
As said, if you like the idea, I will open a new issue and work on an implementation for modularizing this to see if it makes sense 😄